### ISSUE TITLE
Try to understand the usage of baseurl

### ISSUE THREAD

**Issue by jugmac00** [jugmac00/flask-reuploaded#24]
From the docstring:
```
    :param base_url: The URL (ending with a /) that files can be downloaded
                     from. If this is `None`, Flask-Reuploaded will serve the
                     files itself.
```
Does this mean the file is automatically available through the web? I don't think so.

**Comment by jugmac00** [comment]
Yes. It is - and I do not think this is good.

There was an issue on the `Flask-Uploads` issue tracker, which was closed - without an explanation:
https://github.com/maxcountryman/flask-uploads/issues/19

The linked video shows a bad misconfiguration, but there is a point - the auto-serving of the directory or even the files is surprising:
https://www.youtube.com/watch?v=jv3BH3PW-iE

**Comment by Abdur-rahmaanJ** [comment]
@jugmac00 I prefer my own url schemes. How about adding an flag to prevent autoserve?

**Comment by jugmac00** [comment]
@Abdur-rahmaanJ Just yesterday I thought about this issue again, and yes, a flag is a very good idea.

Though, I do not know yet whether autoserver should be opt-in or opt-out.

Opt-in is more secure, opt-out is backwards compatible to `flask-uploads`, which this project tries to be.

What is your take on this?

I need a bit time to think about it.

Thanks for your suggestion!

**Comment by Abdur-rahmaanJ** [comment]
I have my own url schemes specifically to control permissions.

Unless we can extend a mixin or rewrite a function and add some 
decorators on top to control permission, i suggest to drop the 
feature!

Which is a bit harsh. I then suggest disable autoserve by default.

As time passes breaking changes would be needed. Like the werkzeug
issue. As long as the changes are minimal and needed i think it's worth 
it. As, if someone was using flask-uploads, they would have needed to
implement these changes raw, which is why this package exists.

**Comment by tassaron2** [comment]
Discovered this issue as I came here to file a similar issue.

https://github.com/jugmac00/flask-reuploaded/blob/96563f446989ae9dfaa6fb0925a6b7b82f7e2efa/src/flask_uploads/flask_uploads.py#L61

According to this line, the configuration expects `UPLOADED_(name of image set.upper())_URL` to be defined or else it will register its own blueprint for the _uploads endpoint. I can confirm that adding such a config to my Flask object (with an empty string) disables autoserve

There should be a more intuitive way to do it, though. Autoserve also isn't documented anywhere outside the code, as far as I saw.

**Comment by jugmac00** [comment]
@tassaron2 I really appreciate your input!

While my main intention to create `Flask-Reuploaded` was to give the thousands of `Flask-Uploads` users an easy way to keep their app running, and thus I am very cautious to break existing behavior, I have to agree that the documentation is lacking some love. Also, the autoserve *feature* - if kept - has to be documented well, as it could lead to unexpected behavior (e.g. exposure of data which was not meant to be public).

I'd be very happy about a pull request to improve the current documentation, but also to ideas on how to make the configuration more straight forward and more safe, ideally without breaking backwards compatibility much. Though, in this case it may be worth it. Then again, it has to be documented very well, so users migrating from `Flask-Uploads` to `Flask-Reuploaded` have a good experience.

I really like what  @greyli did for Bootstrap ( https://bootstrap-flask.readthedocs.io/en/stable/migrate.html ).

**Comment by Abdur-rahmaanJ** [comment]
I vouch for better docs as the feature does exist

**Comment by jugmac00** [comment]
Hi, thank you once again for you feedback.

I had some spare time and started a pr to fix this issue, see #71 - feedback appreciated!
